collated/main/A0139661Y.md

```diff
@@ -0,0 +1,1790 @@
+                new Detail(details),
+                new DueByDate (dueByDateStart, dueByDateEnd),
+                new DueByTime(dueByTimeStart, dueByTimeEnd),
+                new Priority(priority),
+                new UniqueTagList(tagSet)
+        );
+        isUndoable = true;
+    }
+
+
+    public AddCommand(Task toAdd) {
+        this.toAdd = toAdd;
+    }
+
+    public ReadOnlyTask getTask() {
+        return toAdd;
+    }
+
+    @Override
+    public CommandResult execute() {
+        Blocker blocker = new Blocker();
+    	try {
+    		// Check for block conflicts. Throws a TaskBlockedException if conflicts exist.
+    		blocker.checkBlocked(toAdd, model.getBlockedList());
+        	model.addTask(toAdd);
+            return new CommandResult(String.format(MESSAGE_SUCCESS, toAdd));
+        } catch (TaskBlockedException tbe) {
+    		return new CommandResult(tbe.getMessage());
+        }
+    }
+}
+```
+###### /java/seedu/cmdo/logic/commands/StorageCommand.java
+``` java
+public class StorageCommand extends Command {
+	
+    public static final String COMMAND_WORD = "storage";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Changes the storage folder path of CMDo.\n"
+            + "Parameters: <file/path/name>"
+            + "Example: " + COMMAND_WORD
+            + " /home/Documents";
+
+    public static final String MESSAGE_SUCCESS = "cmdo.xml now saves to %1$s";
+
+    private final String filePath;
+
+    /**
+     * Created an command to change storage filepath.
+     *
+     * @throws IllegalValueException if any of the raw values are invalid
+     * 
+```
+###### /java/seedu/cmdo/logic/commands/StorageCommand.java
+``` java
+     */
+    public StorageCommand(String filePath) {
+    	this.filePath = filePath;
+    }
+
+    @Override
+    public CommandResult execute() {
+    	model.changeStorageFilePath(filePath);
+        EventsCenter.getInstance().post(new StorageFileChangedEvent(filePath));
+        return new CommandResult(String.format(MESSAGE_SUCCESS, filePath));
+    }
+}
+```
+###### /java/seedu/cmdo/logic/LogicManager.java
+``` java
+    @Override
+    public ObservableList<ReadOnlyTask> getBlockedList() {
+    	return model.getBlockedList();
+    }
+    
+```
+###### /java/seedu/cmdo/logic/LogicManager.java
+``` java
+    @Override
+    public ObservableList<ReadOnlyTask> getAllTaskList() {
+    	return model.getAllTaskList();
+    }
+    
+    @Override
+    public ObservableList<ReadOnlyTask> getFilteredTaskList() {
+        return model.getFilteredTaskList();
+    }
+    
+```
+###### /java/seedu/cmdo/logic/LogicManager.java
+``` java
+    @Override
+    public ObservableList<ReadOnlyTask> getFilteredTaskList(boolean firstRun) {
+        return model.getFilteredTaskList(firstRun);
+    }    
+}
+```
+###### /java/seedu/cmdo/logic/parser/Blocker.java
+``` java
+	public ArrayList<LocalDateTime> checkBlocked(Task toCheck, UnmodifiableObservableList<ReadOnlyTask> blockedList)
```

Comment:
Avoid lines longer than 110 chars. When a line is too long, it requires horizontal scrolling, that slows down the reader/reviewer.